### PR TITLE
feat: end method

### PR DESCRIPTION
**Pull request by jgoux**
## What kind of change does this PR introduce?

This PR introduces a new `end` method to `PostgresConnection`, allowing us to have more control over socket termination.

## What is the current behavior?

Currently if we end the socket manually in some of the `onX` options, for example `onTlsUpgrade`, we end up receiving new messages while the socket is already closed, which causes errors like `ERR_STREAM_WRITE_AFTER_END`.

## What is the new behavior?

The user can now call `connection.end()` which will gracefully handle the socket by removing all the internal listeners before ending the socket. It means that we won't process any further messages.

## Additional context

As a user I could also call `connection.detach()` before ending the socket, but it's not super intuitive. `PostgresConnection` should be seen as a smart `net.Socket`, aware of the Postgres wire protocol, so it makes sense to me to expose this `end` method.

### CODE REVIEW

File: .vscode/settings.json
Comment (jgoux) [inline .vscode/settings.json:1]: Just setting the biomejs extension as the default formatter as we're using Biome in the repo 👍

File: packages/pg-gateway/src/connection.ts [hunk 1]
```diff
@@ -117,10 +117,8 @@ export type PostgresConnectionOptions = {
 };
 
 export type MessageResponse =
-  | undefined
-  | Uint8Array
-  | Iterable<Uint8Array>
-  | AsyncIterable<Uint8Array>;
+  // biome-ignore lint/suspicious/noConfusingVoidType: this is a return type
+  void | Uint8Array | Iterable<Uint8Array> | AsyncIterable<Uint8Array>;
```
Comment (jgoux) [inline packages/pg-gateway/src/connection.ts:121]: I fixed the type here: undefined -> void.
Without this fix we were forced to `return undefined;` vs `return;`.
The biome error is only for types that aren't used as a return type, so we can safely ignore it here.

**Review by gregnr — APPROVED** [review]
